examples/step-61/step-61.cc

```diff
@@ -672,10 +669,14 @@ namespace Step61
   template <int dim>
   void WGDarcyEquation<dim>::compute_velocity_errors()
   {
-    const FE_RaviartThomas<dim> fe_rt(fe.base_element(0).degree);
+    const FE_DGRaviartThomas<dim> fe_dgrt(fe.base_element(0).degree);
+    DoFHandler<dim>               dof_handler_dgrt(triangulation);
+    dof_handler_dgrt.distribute_dofs(fe_dgrt);
+
+    darcy_velocity.reinit(dof_handler_dgrt.n_dofs());
```

Comment:
If you move the `dof_handler_dgrt` into this function, then you also need to move the declaration of the `darcy_velocity` vector here because it references the enumeration scheme of the `dof_handler_dgrt`. Is that possible, or are you using the vector in some other function as well?

Reply:
I checked the convergence rates, they do not change.
I calculated darcy_velocity here, and I draw the Darcy velocity figures in output_results.

Reply:
Ah, then we can't move the `dof_handler_dgrt`, I'm afraid. Can you move it back into the class like you had before?

Reply:
Is that possible we also define dof_handler_dgrt in output_results?

Reply:
That's what you are already doing right now.

What I mean is this: Imagine that you decide to use a different enumeration scheme for the `dof_handler_dgrt` in the two places where you define it right now, for example by calling one of the `DoFRenumbering` functions in one of the two places. Because you save the `darcy_velocity` in a member variable, that would mean that you interpret which degrees of freedom the individual components of `darcy_velocity` are associated with using the numbering scheme in the first of these functions, but that will no longer be correct in the second function because you have changed the numbering scheme in between. Does that make sense?

The solutions are either to move the `dof_handler_dgrt` back into a member variable as you had before, or to also make `darcy_velocity` a local variable in each of the two functions -- but then you have to re-compute `darcy_velocity` in `output_results()`. One could do that, but it requires copying a bunch of code and that seems wasteful.

Reply:
Oh. I see. I will change it back to what we had before.